coderd/database/queries/insights.sql

```diff
@@ -22,18 +22,18 @@ GROUP BY workspace_agent_stats.user_id, users.username, users.avatar_url
 ORDER BY user_id ASC;
 
 -- name: GetTemplateInsights :one
--- GetTemplateInsights has a granularity of 5 minutes where if a session/app was
--- in use, we will add 5 minutes to the total usage for that session (per user).
-WITH d AS (
-	-- Subtract 1 second from end_time to avoid including the next interval in the results.
-	SELECT generate_series(@start_time::timestamptz, (@end_time::timestamptz) - '1 second'::interval, '5 minute'::interval) AS d
-), ts AS (
+-- GetTemplateInsights has a granularity of 1 minute where if a session/app was
+-- in use during a minute, we will add 1 minute to the total usage for that
+-- session/app (per user).
+WITH ts AS (
 	SELECT
 		d::timestamptz AS from_,
-		(d + '5 minute'::interval)::timestamptz AS to_,
-		EXTRACT(epoch FROM '5 minute'::interval) AS seconds
-	FROM d
-), usage_by_user AS (
+		(d::timestamptz + '1 minute'::interval) AS to_,
+		EXTRACT(epoch FROM '1 minute'::interval) AS seconds
+	FROM
+		-- Subtract 1 second from end_time to avoid including the next interval in the results.
```

Comment:
Correct me if I'm wrong, but for 7 days, it will give 7 * 24 * 60 records per every JSON response? If there is really low usage, there will be a lot of empty slots. If this is a valid concern, maybe we should minimize the response.

Reply:
It's a fair point, it will slow the query down a bit. I'd need to test against a real-ish dataset to figure out by how much. For very little data the overhead of 1m vs 5m is not too bad at ~65ms vs ~35ms, but that's just for one week.

Reply:
I've kept both at 5 minute roundups now for speed (trade-off for correctness).